crates/bevy_window/src/window.rs

```diff
@@ -496,355 +538,264 @@ impl Window {
-        });
-    }
-    /// Get whether or not the window is always on top.
-    #[inline]
-    pub fn always_on_top(&self) -> bool {
-        self.always_on_top
-    }
+/// The different states a window can be in.
+#[derive(Default, Component, Debug, Copy, Clone, PartialEq, Eq, Reflect)]
+#[reflect(Component)]
+pub enum WindowState {
+    /// The window is neither maximized nor minimized. Usually the
+    /// default state of a window.
+    #[default]
+    Normal,
+
+    /// The window is minimized to the task bar, but the program is
+    /// still running.
+    Minimized,
+
+    /// The window is taking up the maximum amount of space on the
+    /// window it is allowed to, without becoming fullscreen.
+    Maximized,
+}
 
-    /// Set whether of not the window is always on top.
-    pub fn set_always_on_top(&mut self, always_on_top: bool) {
-        self.always_on_top = always_on_top;
-        self.command_queue
-            .push(WindowCommand::SetAlwaysOnTop { always_on_top });
+impl WindowState {
+    /// Returns true if [`WindowState`] is Normal.
+    pub fn normal(&self) -> bool {
+        self == &Self::Normal
     }
-    /// Close the operating system window corresponding to this [`Window`].
-    ///  
-    /// This will also lead to this [`Window`] being removed from the
-    /// [`Windows`] resource.
-    ///
-    /// If the default [`WindowPlugin`] is used, when no windows are
-    /// open, the [app will exit](bevy_app::AppExit).  
-    /// To disable this behaviour, set `exit_on_all_closed` on the [`WindowPlugin`]
-    /// to `false`
-    ///
-    /// [`Windows`]: crate::Windows
-    /// [`WindowPlugin`]: crate::WindowPlugin
-    pub fn close(&mut self) {
-        self.command_queue.push(WindowCommand::Close);
-    }
-    #[inline]
-    pub fn drain_commands(&mut self) -> impl Iterator<Item = WindowCommand> + '_ {
-        self.command_queue.drain(..)
+
+    /// Returns true if [`WindowState`] is Maximized.
+    pub fn maximized(&self) -> bool {
+        self == &Self::Maximized
     }
-    /// Get whether or not the window has focus.
-    ///
-    /// A window loses focus when the user switches to another window, and regains focus when the user uses the window again
-    #[inline]
-    pub fn is_focused(&self) -> bool {
-        self.focused
+
+    /// Returns true if [`WindowState`] is Minimized.
+    pub fn minimized(&self) -> bool {
+        self == &Self::Minimized
     }
-    /// Get the [`RawHandleWrapper`] corresponding to this window if set.
-    ///
-    /// During normal use, this can be safely unwrapped; the value should only be [`None`] when synthetically constructed for tests.
-    pub fn raw_handle(&self) -> Option<RawHandleWrapper> {
-        self.raw_handle.as_ref().cloned()
+}
+
+/// Defines how this window should behave in relation to web-canvas elements.
+#[derive(Default, Component, Debug, Clone, PartialEq, Eq, Reflect)]
+#[reflect(Component)]
+pub struct WindowCanvas {
```

Comment:
This should probably be behind a platform-specific feature.

Reply:
I think it might be better as a non-platform specific feature so that you can define a `WindowBundle` and it should just work on web without having to fidget with feature flags.